crates/rollup-boost/src/health.rs

```diff
@@ -35,7 +54,9 @@ impl HealthHandle {
                     Ok(block) => block,
                     Err(e) => {
                         warn!(target: "rollup_boost::health", "Failed to get unsafe block from builder client: {} - updating health status", e);
-                        self.probes.set_health(Health::PartialContent);
+                        if !self.execution_mode.lock().is_disabled() {
```

Comment:
I feel dry-run mode should work the same way as disabled since the goal there is also to just rely on L2 block-building while seeing how the builder client behaves when receiving engine calls

What do you think about having PartialContent being returned only when execution_mode=enabled?